sqlglot/optimizer/scope.py

```diff
@@ -231,6 +236,17 @@ def subqueries(self):
         self._ensure_collected()
         return self._subqueries
 
+    @property
+    def stars(self):
```

Comment:
why did you need to add this?

Reply:
The `scope.columns` wouldn't collect stars at all before so we couldn't recognize, annotate etc these expressions. Check out [this branch](https://github.com/tobymao/sqlglot/tree/jo/refactor_dot_conversion_in_qualify) which is the foundation of this PR

Reply:
Just adding a bit to what Vaggelis said:

> wouldn't collect stars

FYI, relevant section in `Scope._collect`: https://github.com/tobymao/sqlglot/blob/main/sqlglot/optimizer/scope.py#L128-L129.

TL;DR we'd never convert columns to dots in `_qualify_columns`, because it happens inside the `scope.columns` loop and stars would not be visited.